src/main/scala/com.snowplowanalytics.snowplow.analytics.scalasdk/ParsingError.scala

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright (c) 2016-2019 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package com.snowplowanalytics.snowplow.analytics.scalasdk
+
+import cats.data.NonEmptyList
+import cats.syntax.either._
+import io.circe._
+import io.circe.syntax._
+import com.snowplowanalytics.snowplow.analytics.scalasdk.decode.Key
+
+/**
+  * Represents error during parsing TSV event
+  */
+sealed trait ParsingError
+
+object ParsingError {
+
+  /**
+    * Represents error which given line is not a TSV
+    */
+  final case object NonTSVPayload extends ParsingError
+
+  /**
+    * Represents error which number of given columns is not equal
+    * to number of expected columns
+    * @param columnCount mismatched column count in the event
+    */
+  final case class ColumnNumberMismatch(columnCount: Int) extends ParsingError
+
+  /**
+    * Represents error which encountered while decoding values in row
+    * @param errors Infos of errors which encountered during decoding
+    */
+  final case class RowDecodingError(errors: NonEmptyList[RowDecodingErrorInfo]) extends ParsingError
+
+  /**
+    * Gives info about the reasons of the errors during decoding value in row
+    */
+  sealed trait RowDecodingErrorInfo
+
+  object RowDecodingErrorInfo {
+    /**
+      * Represents cases where value in a field is not valid
+      * e.g invalid timestamp, invalid UUID
+      * @param key key of field
+      * @param value value of field
+      * @param message error message
+      */
+    final case class InvalidValue(key: Key, value: String, message: String) extends RowDecodingErrorInfo
+
+    /**
+      * Represents cases which getting error is not expected while decoding row
+      * For example, while parsing the list of tuples to HList in the
+      * RowDecoder, getting more or less values than expected is impossible
+      * due to type check. Therefore 'UnexpectedRowDecodingError' is returned for
+      * these cases. These errors can be ignored since they are not possible to get
+      * @param error error message
+      */
+    final case class UnexpectedRowDecodingError(error: String) extends RowDecodingErrorInfo
```

Comment:
I don't think `Unexpected` here is descriptive, it is rather `Unhandled` or `Undefined`.